### PR TITLE
Classes

### PR DESCRIPTION
**Pull request by chthota9**

### CODE REVIEW

**Review by tim-nguyen-cs — APPROVED** [review]
All changes look good to go!

File: webApp/server_modules/routes/profile.js [hunk 1]
```diff
@@ -138,7 +117,35 @@ router.post('/user/:id(\\d+)/review/:course(\\d+)/sub', (req, res) => {
...
 
 // A route that will actually create a user's account within the database
+router.post('/createProfile', function(req, res) {
+    let profile = newProfile(req.body, req.user.id, req.user.extra);
+    addUser(profile)
+        .then(profile => {
+            req.login({ id: profile.googleID }, err => {
+                if (err) return res.redirect('/');
+                res.redirect('/');
+            });
```
Comment (noll115) [inline webApp/server_modules/routes/profile.js:127]: Everything after this line might cause an error due to `res.redirect()` being before it all . A possible fix would be moving all tutor adding functionality before `req.login()`
```suggestion
'insert the tutor adding functionality here once done then do the rest'
    req.login(....
        res.redirect('/');
            });
```

File: webApp/server_modules/routes/profile.js [hunk 2]
```diff
@@ -149,10 +156,47 @@ router.post('/updateProfile', function(req, res) {
     updateUser(req.user, req.body)
         .then(() => res.json({ sucessful: true }))
         .catch(() => res.json({ sucessful: false }));
+
```
Comment (noll115) [inline webApp/server_modules/routes/profile.js:159]: The same problem appears here with `res.json()` executing before they are added or removed from the courses as a tutor

**Review by noll115 — CHANGES_REQUESTED** [review]
`/createProfile` and `/updateProfile` routes will return before the page before actually adding the account to the classes they are teaching

**Review by noll115 — APPROVED** [review]

**Pull request merged**